### PR TITLE
Freelancer platoon part 1, the basics

### PR DESCRIPTION
**Pull request by Merrgear**
# About the pull request

Adds a platoon of freelancer mercs. lays out the groundwork for further additions (better sprites, more lore, maybe even return to OG gunny weapon plan of FLASH). 

Major features include: Fuckton of stolen or bought weaponry in the armory, not too much ammo of one type to simulate the feel of ammo scarcity among the mercs, but not actual little amount of ammo so you arent fucked if an unnamed person takes 16 mags every round. 

Lack of PlatCo, mercs wouldnt really have an overwatch system AND if they had a real "officer" type it would be the captain, and then who would the GMs have to give orders from and call about situation and such. replaced with totally original RTO idea i have not taken from the tried and tested FORECON platoon.

New shipmap, not much to say about it im not the best mapper.

A single squad, the idea is it adds to the unmanaged hoodlum aspect, but im willing to fold if its too much trouble player wise to manage (will see if its the case if it ever gets a playtest), they still get gunny and a sqsgt so they can always divy up ingame between them

# Explain why it's good for the game

More options for GMs to do missions, more RP opportunity for the players in a VERY different than all current platoons platoon. Bane can complete pirate larp

# Changelog

:cl:
add: Added FL platoon and all its goodies
add: Framework for a secondary gunny weapon to replace current flamer in part 2 if it works
mapadd: added the SS Corsair, the totally not shady civilian transport vessel
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/rifles.dm [hunk 1]
```diff
@@ -1705,6 +1807,62 @@
 	current_mag = /obj/item/ammo_magazine/rifle/l42a/rubber
 
 //-------------------------------------------------------
+//L42A Battle Rifle FAL ripoff edition
+
+/obj/item/weapon/gun/rifle/l42B
```
Comment (AndroBetel) [inline code/modules/projectiles/guns/rifles.dm:1812]: this too

File: code/modules/projectiles/guns/smgs.dm [hunk 2]
```diff
@@ -415,6 +456,25 @@
 	damage_mult = BASE_BULLET_DAMAGE_MULT + BULLET_DAMAGE_MULT_TIER_3
 	recoil_unwielded = RECOIL_AMOUNT_TIER_5
 
+/obj/item/weapon/gun/smg/bizonlancer
```
Comment (AndroBetel) [inline code/modules/projectiles/guns/smgs.dm:459]: erm you get what i mean

File: code/game/machinery/vending/vendor_types/squad_prep/squad_prep.dm [hunk 3]
```diff
@@ -307,7 +307,84 @@
...
+/obj/structure/machinery/cm_vending/sorted/uniform_supply/squad_prep/fl
+	name = "\improper ColMarTech Freelancer Uniform Vendor"
 
+/obj/structure/machinery/cm_vending/sorted/uniform_supply/squad_prep/fl/populate_product_list(scale)
+	listed_products = list(
+		list("STANDARD EQUIPMENT", -1, null, null, null),
+		list("Combat Boots", round(scale * 15), /obj/item/clothing/shoes/marine/knife, VENDOR_ITEM_REGULAR),
+		list("Freelancer Uniform", round(scale * 15), /obj/item/clothing/under/marine/veteran/freelancer, VENDOR_ITEM_REGULAR),
+		list("Combat Gloves", round(scale * 15), /obj/item/clothing/gloves/marine, VENDOR_ITEM_REGULAR),
+		list("RC6 Helmet", round(scale * 15), /obj/item/clothing/head/helmet/marine/veteran/ua_riot, VENDOR_ITEM_REGULAR),
```
[the remark this answers is not in the document]
Comment (Merrgear) [reply code/game/machinery/vending/vendor_types/squad_prep/squad_prep.dm:319]: only real things that would be in their hands and that also prevents IBs that is currently in code. i want to make sure the actual framework works before i go making new sprites for new objects and all that jazz

**Comment by FslashN** [comment]
So, question: We have a bunch of ragtag guns in the code already that are overall worse than faction guns but do a decent enough job. Why all the new weapons?

**Comment by Merrgear** [comment]
> So, question: We have a bunch of ragtag guns in the code already that are overall worse than faction guns but do a decent enough job. Why all the new weapons?

the /lancer variants fix bizon and the ppsh 43 to not have the spread of a shotgun blast. the /repro variants add full auto, remove starting random attachments, and update the lore to not be weird out of place guns from the past in the future. the lancer variant of the minigun has updated lore. and the L42B contains full auto, separate bullets from pulse so you cant just find a single pulse rifle mag and be set for a while, and an updated sound to not be as weak with the full auto. more FAL like L42 variant

**Pull request closed without merging**